rust/src/io/reader.rs

```diff
@@ -754,6 +823,66 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn read_with_delete() {
+        let arrow_schema = ArrowSchema::new(vec![
+            ArrowField::new("i", DataType::Int64, true),
+            ArrowField::new("f", DataType::Float32, false),
+        ]);
+        let schema = Schema::try_from(&arrow_schema).unwrap();
+
+        let store = ObjectStore::memory();
+        let path = Path::from("/foo");
+
+        // Write 10 batches.
+        let mut file_writer = FileWriter::try_new(&store, &path, schema.clone())
+            .await
+            .unwrap();
+        for batch_id in 0..10 {
+            let value_range = batch_id * 10..batch_id * 10 + 10;
+            let columns: Vec<ArrayRef> = vec![
+                Arc::new(Int64Array::from_iter(
+                    value_range.clone().collect::<Vec<_>>(),
+                )),
+                Arc::new(Float32Array::from_iter(
+                    value_range.map(|n| n as f32).collect::<Vec<_>>(),
+                )),
```

Comment:
Shouldn't need `collect::<Vec<_>>` here.
```suggestion
                Arc::new(Int64Array::from_iter(
                    value_range.clone(),
                )),
                Arc::new(Float32Array::from_iter(
                    value_range.map(|n| n as f32),
                )),
```

Could also do:


```rust
Arc::new(value_range.clone().collect::<Int64Array>())
```